lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -2429,10 +2406,47 @@ export class UserAgentApplication {
    */
   private getHostFromUri(uri: string): string {
     // remove http:// or https:// from uri
-    var extractedUri = String(uri).replace(/^(https?:)\/\//, "");
+    // TODO: Test this:: return  extractedUri = String(uri).replace(/^(https?:)\/\//, "").split("/")[0];
+    let extractedUri = String(uri).replace(/^(https?:)\/\//, "");
     extractedUri = extractedUri.split("/")[0];
     return extractedUri;
   }
 
+  /**
+   * Utils function to create the Authentication
+   * @param userObject
+   * @param scopes
+   * @param silentCall
+   */
+  private getTokenType(userObject: User, scopes: string[], silentCall: boolean) {
+
+    // if user is passed and matches the user object/or set to getUser() from cache
+    // if client-id is passed as scope, get id_token else token/id_token_token (in case no session exists)
+    let tokenType;
+
+    // acquireTokenSilent
+    if (silentCall) {
+      if (Utils.compareObjects(userObject, this.getUser())) {
+        tokenType = (scopes.indexOf(this.clientId) > -1) ? ResponseTypes.id_token : ResponseTypes.token;
```

Comment:
array.includes

Reply:
scopes is a string[]. There is an issue with this with es5 explained here: https://github.com/Microsoft/TypeScript/issues/11890

Currently our tsconfig is:    
 "target": "es5",
  "lib": [ "es2015", "dom", "es2015.promise" ]

We would need:
"lib": [
  "dom",
  "es7"
]
for 'include' to work on string[]. Though I agree it looks better, I do not want to make compilation changes for this at this point. Hence refraining for now.

Lets sort the library generation and compatibility for all es* versions and then revisit this.